core/arch/arm32/kernel/thread.c

```diff
@@ -73,6 +148,61 @@ thread_pm_handler_t thread_system_reset_handler_ptr;
 
 static unsigned int thread_global_lock = UNLOCK;
 
+#ifdef CFG_WITH_STACK_CANARIES
+static void init_canaries(void)
+{
+	size_t n;
+#define INIT_CANARY(name)						\
+	for (n = 0; n < ARRAY_SIZE(name); n++) {			\
+		uint32_t *start_canary = &GET_START_CANARY(name, n);	\
+		uint32_t *end_canary = &GET_END_CANARY(name, n);	\
+									\
+		*start_canary = START_CANARY_VALUE;			\
+		*end_canary = END_CANARY_VALUE;				\
+		DMSG("#Stack canaries for %s[%zu] with top at %p\n",	\
+			#name, n, (void *)(end_canary - 1));		\
```

Comment:
`end_canary + 1` is not correct. The stacks grows from high address to low address.